ebnf/main.go

```diff
@@ -0,0 +1,96 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+)
+
+type scanner struct {
+	input []byte
+	ch    byte
+	pos   int
+
+	insideBraces int
+}
+
+func (s *scanner) scan() {
```

Comment:
The EBNF grammar is very similar to the YACC grammar. 3 things needs to be changed:
- `:` becomes `::=`
- semicolon (`;`) is not part of EBNF
- We have to remove the Go code inside braces `{}`

This main logic is to scan byte by byte the grammar file considering the rules above.

Reply:
To illustrate how the EBNF looks like, here are the first rules:

````
start ::= stmts 

stmts ::=  
  single_stmt semicolon_opt
  | multi_stmts semicolon_opt

single_stmt ::= 
  select_stmt
  | create_table_stmt

multi_stmts ::= 
  multi_stmt 
  | multi_stmts ';' multi_stmt
```

Reply:
You can check how the above grammar looks like in https://www.bottlecaps.de/rr/ui